### ISSUE TITLE
Improve error handling for incorrect raw parameters

### ISSUE THREAD

**Issue by timsuchanek** [prisma/prisma#2561]
Since [Beta 6](https://github.com/prisma/prisma/releases/tag/2.0.0-beta.6), we allow parameters in `.raw` like so:

```ts
const result = await prisma.raw(
  'SELECT * FROM User WHERE id = $1 OR email = $2;',
  1,
  'e@ma.il'
)
```

While we chose an API, where each parameter is a separate function arg for `.raw`, the API could also have looked like this:

```ts
const result = await prisma.raw(
  'SELECT * FROM User WHERE id = $1 OR email = $2;',
  [1, 'e@ma.il']
)
```

However, if I provide that, I don't get a lot of help but a PANIC:
https://github.com/prisma/prisma-client-js/issues/705

**Comment by pimeys** [comment]
https://github.com/prisma/prisma-client-js/issues/705#issuecomment-635386345

**Comment by pimeys** [comment]
https://github.com/prisma/quaint/issues/137

**Comment by pimeys** [comment]
Fixed in prisma/prisma-engines#827
